controllers/clusters/cassandra_controller.go

```diff
@@ -385,6 +385,34 @@ func (r *CassandraReconciler) handleUpdateCluster(
 		}
 	}
 
+	addedRefs, deletedRefs := cassandra.Status.AvailableUsers.Diff(cassandra.Spec.UserRefs)
+
+	for _, ref := range addedRefs {
+		err = r.handleUsersCreate(ctx, l, cassandra, ref)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	for _, ref := range deletedRefs {
+		err = r.handleUsersDelete(ctx, l, cassandra, ref)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	}
+
+	if addedRefs != nil || deletedRefs != nil {
+		cassandra.Status.AvailableUsers = cassandra.Spec.UserRefs
+		err = r.Status().Patch(ctx, cassandra, patch)
+		if err != nil {
+			l.Error(err, "Failed to patch cluster status with created user")
+			r.EventRecorder.Eventf(cassandra, models.Warning, models.CreationFailed,
+				"Failed to create a user for the cluster. Reason: %v", err)
+
+			return reconcile.Result{}, err
+		}
+	}
```

Comment:
if possible please make it generic and call it `handleUserChanges`

Reply:
I've implemented the following flow as a generic by using interfaces. Also, I've implemented a generic func for detaching users from the cluster.